internal/publish/interpreters.go

```diff
@@ -34,6 +34,10 @@ func (p *defaultPublisher) configureInterpreters() error {
 		if err != nil {
 			return err
 		}
+		// Enforce modern renv.lock format when publishing
+		if err := renv.ValidateModernLockfile(lockfile); err != nil {
+			return err
+		}
```

Comment:
In the end this is not strictly required, but to make things more consistent it is better to restrict renv.lock format to >= 1.1.0 it's a safe trade-off, asking users to only update renv seems reasonable.